### PR TITLE
contrib: add extra cern data

### PR DESCRIPTION
**Pull request by zzacharo**

### CODE REVIEW

File: invenio_oauthclient/contrib/cern.py [hunk 1]
```diff
@@ -258,8 +258,11 @@ def account_info(remote, resp):
     """Retrieve remote account information used to find local user."""
     resource = get_resource(remote)
 
+    person_id = resource.get('PersonID', [None])
+    external_id = resource.get('uidNumber', person_id)
+
     email = resource['EmailAddress'][0]
-    external_id = resource['uidNumber'][0]
+    external_id = external_id[0]
```
Comment (ntarocco) [inline invenio_oauthclient/contrib/cern.py]: simply
```
resource.get('uidNumber', person_id[0])
```
?

**Comment by lnielsen** [comment]
Ping @tiborsimko @pamfilos - Please check this PR as you are using the CERN OAuth contrib.

File: tests/test_contrib_cern.py [hunk 2]
```diff
@@ -37,6 +38,22 @@ def test_fetch_groups(app, example_cern):
                for group in ('Group{}'.format(i) for i in range(4, 6)))
 
 
+def test_fetch_extra_data(app, example_cern):
```
Comment (ntarocco) [inline tests/test_contrib_cern.py]: Can you please add a test where the response does NOT contain some of the values too?
So we ensure that future changes will not break behaviour. Thanks!

File: invenio_oauthclient/contrib/cern.py [hunk 3]
```diff
@@ -206,9 +220,12 @@ def account_groups(account, resource, refresh_timedelta=None):
         return account.extra_data.get('groups', [])
 
     groups = fetch_groups(resource['Group'])
+    extra_data = fetch_extra_data(resource)
```
Comment (zzacharo) [inline invenio_oauthclient/contrib/cern.py]: Also @lnielsen @pamfilos @tiborsimko do you find any case that this should be configurable in order in the future to parse any field in the cern oauth response?

File: tests/test_contrib_cern.py (hunk 2 above)
Comment (zzacharo) [reply tests/test_contrib_cern.py]: @ntarocco [here](https://github.com/inveniosoftware/invenio-oauthclient/pull/179/files#diff-25fc83081804f5e108a7dd26baf648beR57)

**Comment by ntarocco** [comment]
I think this is closing https://github.com/inveniosoftware/invenio-oauthclient/issues/178 if you want to mention it

**Comment by zzacharo** [comment]
@ntarocco no actually. I reverted the fix and I am gonna create a separate pr for that.

**Comment by zzacharo** [comment]
Ping @lnielsen @tiborsimko @pamfilos could you please check it as I need this?

File: invenio_oauthclient/contrib/cern.py (hunk 3 above)
Comment (pamfilos) [reply invenio_oauthclient/contrib/cern.py]: @zzacharo I think since we are adding this feature, it's better to have it configurable

**Comment by egabancho** [comment]
@zzacharo do you think we should write an upgrade recipe or something to update/invalidate all the accounts which don’t have this data?  Perhaps simply force to reload it using the last updated field.

**Comment by zzacharo** [comment]
@egabancho yes that would be good I think. @lnielsen I don't think it will be incompatible especially if we write a recipe to update user's metadata with what @egabancho proposed..I don't know if I miss something...

**Comment by ntarocco** [comment]
@egabancho the refresh time is a config of 5 minutes by default (if I am not wrong), I don't think any of us changed it? At least nor videos or cern analysis preservation...
@lnielsen I also don't think it is breaking since it is adding new fields and not changing existings...

**Pull request merged**